framework/graphics/dx12_util.h

```diff
@@ -187,16 +187,22 @@ bool GetAdapterAndIndexbyDevice(ID3D12Device*                     device,
 // Get the adapter at specified index
 IDXGIAdapter* GetAdapterbyIndex(graphics::dx12::ActiveAdapterMap& adapters, int32_t index);
 
+// Returns whether the device passed represents a unified memory architecture
+// (UMA) GPU as with most integrated graphics / APUs and mobile SOCs.
+// If the underlying D3D call fails, the function will also return false and the
+// error will be logged.
+bool IsUma(ID3D12Device& device);
+
```

Comment:
@davidd-lunarg Is this a good place for this little helper function and are you okay with the bool I added to those below?

Reply:
Yes looks good.